jcl/src/jdk.attach/share/classes/com/ibm/tools/attach/attacher/OpenJ9VirtualMachine.java

```diff
@@ -117,6 +127,22 @@
 	 *             if the descriptor is null the target does not respond.
 	 */
 	void attachTarget() throws IOException, AttachNotSupportedException {
+		PrivilegedExceptionAction<Object> action = () -> {attachTargetImpl(); return null;};
```

Comment:
There are public methods that need similar changes, such as getThreadInfo(), loadAgentPath(), etc.

Reply:
There is another case:
https://github.com/eclipse/openj9/blob/fde4c8087c1e700ea6d396fa269b4f9aa4d375bb/jcl/src/jdk.attach/share/classes/com/ibm/tools/attach/attacher/OpenJ9VirtualMachine.java#L378

Reply:
That is called from the public APIs, startManagementAgent(), startLocalManagementAgent(), loadAgent(), loadAgentLibrary(), loadAgentPath(). They all (including any other public APIs) need doPriv().

Reply:
`doPriv()` with `Integer.getInteger(status)` within `private static Integer getStatusValue(String response)` is sufficient.
Also notice another occurrence
https://github.com/eclipse/openj9/blob/cbd2a76412c17835db44b2cd0b606f56e58d8f46/jcl/src/java.base/share/classes/com/ibm/tools/attach/target/AttachHandler.java#L58

Will fix them at top of Peter's PR.

Reply:
getInteger() is not the only thing that needs a doPriv(). If you only fix these at a low level, there will be other problems as well with Socket and File operations.

Reply:
Agreed that Socket and File operations require `doPrivileged()` as well. 
The general security guideline is to make the privileged code section as small as possible, so those `Socket` and `File operations` is preferred to be protected within `doPrivileged()` at lower level instead of marking the caller public API themself.

Reply:
It appears https://github.com/eclipse/openj9/blob/fde4c8087c1e700ea6d396fa269b4f9aa4d375bb/jcl/src/jdk.attach/share/classes/com/ibm/tools/attach/attacher/OpenJ9VirtualMachine.java#L378
is an API misusage.

`Integer.getInteger(String nm)` is to "_Determines the integer value of the **system property** with the specified name._"  [1].

As per comments (and context), `private static Integer getStatusValue(String response)` intends to parse the status value from the end of the response string which will be a numeric string at the end of the string. 

It should be `Integer.valueOf(status);` [2]  instead. In such case, there is no need for `doPrivileged()` at all.

[1] https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html#getInteger-java.lang.String- 
[2] https://docs.oracle.com/javase/8/docs/api/java/lang/Integer.html#valueOf-java.lang.String-